lib/hathifiles_database/dumper.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module HathifilesDatabase
+  # TSV export class for hf table and monthly hathifile.
+  class Dumper
+    attr_reader :connection
+
+    def initialize(connection)
+      @connection = connection
+    end
+
+    # Create a TSV dump based on the hf table only.
+    # Used for constructing the delta between a monthly hathifile and the current
+    # state of the database.
+    def dump(output_file:)
+      sql = <<~END_SQL
+        SELECT
+          htid, access, rights_code, bib_num, description, source, source_bib_num,
+          oclc, isbn, issn, lccn, title, imprint, rights_reason,
+          DATE_FORMAT(rights_timestamp, "%Y-%m-%d %H:%i:%s") AS rights_timestamp,
+          us_gov_doc_flag, rights_date_used, pub_place, lang_code, bib_fmt,
+          collection_code, content_provider_code, responsible_entity_code,
+          digitization_agent_code, access_profile_code, author
+        FROM hf
+        ORDER BY htid
+      END_SQL
+
+      File.open(output_file, "w:utf-8") do |file|
+        connection.rawdb.fetch(sql).each do |line|
```

Comment:
Do we have a sense that this is roughly similar performance to if we did `mysql -N -B -e "SELECT all_that FROM hf"`? So long as it's streaming results to the file rather than trying to keep everything in memory it should be fine.

We're already sorting externally (which I think is generally preferable as it reduces load on the database server and is likely to be more performant overall), so let's remove the `ORDER BY htid` clause.

Reply:
The Ruby implementation timed at 319 seconds and the command line version at 146 seconds. But the output of the command line had some string escaping that I would need to figure out (for example the lccn entry for uc1.31822017142498 has a random backslash which Ruby doesn't escape but MySQL/Mariadb does).

Reply:
Tried `--raw` and it worked. The Sequel and the mysql versions create identical output files now.